### DISCUSSION TITLE
How can I add data labels to `Horizontal stacked bar chart`?

### DISCUSSION THREAD

**Discussion by winner106** [observablehq/plot#1781] [Q&A] [answered]
Hello everyone, I have been studying plot recently but have encountered some issues. I hope to get some help. Thank you very much!

I want to add three data labels to this chart, but I couldn't achieve it. The desired effect is as follows:

![image](https://github.com/observablehq/plot/assets/50519642/b8d69d62-151a-4b4f-b9e1-fb6da4478def)

![image](https://github.com/observablehq/plot/assets/50519642/934a166a-8291-422c-bc1f-da23c93fdd50)

[https://observablehq.com/d/6a287381aff76d7f](https://observablehq.com/d/6a287381aff76d7f)

what I have tried:

![image](https://github.com/observablehq/plot/assets/50519642/068af32b-9660-4e26-a7db-5b55f8da17f7)

thank you!

**Comment by Fil** [comment, marked as answer]
here's a solution:

```
Plot.plot({
  marginLeft: 90,
  marks: [
    Plot.barX(
      congress,
      Plot.groupY(
        { x: "count" },
        { fill: "gender", y: "party", sort: { y: "x", reverse: true } }
      )
    ),
    Plot.text(
      congress,
      Plot.stackX(
        Plot.groupY({ x: "count", text: "count" }, { z: "gender", y: "party" })
      )
    )
  ]
})
```

note that you need to pas the **data** before the options, and you need to be explicit about stacking when using the text mark (whereas the bar mark stacks implicitly).

**Reply by winner106** [reply]
Thank you so much. Can I add another total count text at the end of each bar as well?

![image](https://github.com/observablehq/plot/assets/50519642/b89d7733-4d0a-4a4f-b851-8360a2d0c102)

I understand that the "transform" feature can be a bit challenging, but I really want to master it because once you get the hang of it, it can be very convenient for future use. Thank you!

**Reply by Fil** [reply]
Sure! Plot's gallery has an example https://observablehq.com/@observablehq/plot-horizontal-bar-chart-with-label
